Api task

IgorGerasimenko opened the pull request:
Первое ревью кода

IgorGerasimenko committed:
Добавил Апи тест

IgorGerasimenko committed:
Добавил фильтр и проверку

Plodick commented on src/test/java/apiTests/pojos/UserPojo.java:
```diff
@@ -5,19 +5,23 @@
     private int id;
     private String email;
-    @JsonProperty
     private String first_name;
```
Поля класса называй в классическом camelCase для Java. JsonProperty принимает параметры, чтобы соответствие с названиями полей в json'е проставить

Plodick commented on src/test/java/apiTests/pojos/UserPojo.java:
```diff
@@ -5,19 +5,23 @@
 
 @Data
 public class UserPojo {
```
Сам класс можешь обозвать просто User

Plodick commented on src/test/java/apiTests/pojos/UserPojo.java:
```diff
@@ -5,19 +5,23 @@
-}
-
 
+    public String getFirst_Name() {
```
Зачем тебе геттеры, если ты используешь lombok?

Plodick commented on src/test/java/apiTests/Users.java:
```diff
@@ -23,25 +19,43 @@ public void getUsers (){
+                .when().get()
+                .then()
+                .statusCode(200)
+                .extract().jsonPath().getList("data", UserPojo.class);
```
Внимательно прочитай содержание запроса и сделай проход по страницам, чтобы найти Lawson'а.

Plodick commented on src/test/java/apiTests/Users.java:
```diff
@@ -23,25 +19,43 @@ public void getUsers (){
-////                .assertThat().body("email", equalTo("george.bluth@reqres.in") );
-
+    @Test
+    public void checkMichaelLawsonEmailPattern  (){
```
Соблюдай стиль

Plodick commented on src/test/java/apiTests/Users.java:
```diff
@@ -23,25 +19,43 @@ public void getUsers (){
-
-
+                UserPojo result = users.stream()
+                .filter(p -> {
```
Неплохой вариант, но я тебе советую все таки ассерты проставить в соответствии с заданием. Задание гласит так: "Проверить, что у пользователя George Bluth e-mail  george.bluth@reqres.in.". То есть ты должен сначала найти пользователя с соответствующим именем, а потом проверить у него почту. Для чего? Все просто, чтобы было понятно на каком этапе мы свалимся. На поиске юзера или на том, что у него некорректная почта.

Plodick commented on src/test/java/apiTests/Users.java:
```diff
@@ -23,25 +19,43 @@ public void getUsers (){
-
+                UserPojo result = users.stream()
+                .filter(p -> {
+                    if ("george.bluth@reqres.in".equals(p.getEmail()) &
```
Зачем ты ставишь здесь if, если само это выражение может вернуть либо true, либо false.

IgorGerasimenko committed:
Решение задаче через циклы, не самый лучший вариант, но оставлю на память

IgorGerasimenko committed:
Исправил все замечания, кроме "Про ломбок" - геттерры, которые он сгенерил - почему-то недоступны :(
Добавил класс-помощник, вынес туда нетестовые методы

Plodick commented on pom.xml:
```diff
@@ -29,6 +41,40 @@
+        </dependency>
+
+        <dependency>
+            <groupId>org.assertj</groupId>
```
Ты его где-то используешь вообще?

Plodick commented on src/test/java/apiTests/helpers/UserHelper.java:
```diff
@@ -0,0 +1,48 @@
+                .statusCode(200)
+                .extract().jsonPath().getList("data", User.class);
+
+        searchResult = users.stream()
```
Всё хорошо, но не нравятся мне эти статики, в них-то и нужды нет, а если думать глобальнее, то они еще и будут потоконебезопасны, при возможной параллелизации.

Plodick replied:
Пусть методы возвращают что-то конкретное.

Plodick commented on src/test/java/apiTests/helpers/UserHelper.java:
```diff
@@ -0,0 +1,48 @@
+                .extract()
+                .jsonPath();
+
+        pageCount = jspath.getString("total_pages");
```
Данный метод может вполне сразу возвращать целочисленное значение, чтобы потом не конвертить.

Plodick commented on src/test/java/apiTests/Users.java:
```diff
@@ -0,0 +1,57 @@
+                userWasFound = true;
+                System.out.println("Пользователь найден на странице " + pageNumber + ",проверяю его email");
+                assertEquals(searchResult.getEmail(), "michael.lawson@reqres.in");
+                break;
```
Могу предложить в качестве альтернативы выходить из всего метода, тогда строчек будет чуть меньше, а логическая переменная не понадобится.

IgorGerasimenko committed:
-Убрал статические переменные
-Почистил pom.xml
-Методы возвращают, что должны возвращать
-Переменные внутри тестов
-Значение ключа 'total pages' получаю стразу в int
